oldabe/money_in.py

```diff
@@ -557,52 +564,44 @@ def distribute_payment(payment, attributions):
+        amounts_payable = {email: amount
+                           for email, amount in amounts_owed.items()
+                           if email not in unpayable_contributors}
+
+        # use the amount owed to each contributor to draw down any advances
+        # they may already have and then decrement their amount owed accordingly
+        advance_totals = get_sum_of_advances_by_contributor()
         negative_advances = []
         for email, advance_total in advance_totals.items():
-            amount_owed = amounts_owed.get(email, 0)
-            drawdown_amount = max(advance_total - amount_owed, 0)
+            amount_payable = amounts_payable.get(email, 0)
+            drawdown_amount = max(advance_total - amount_payable, 0)
             if drawdown_amount:
                 negative_advance = Advance(email=email,
-                                           amount=-drawdown_amount # note minus sign
-                                           payment_file=payment.payment_file
+                                           amount=-drawdown_amount, # note minus sign
+                                           payment_file=payment.payment_file,
                                            commit_hash=commit_hash)
                 negative_advances.append(negative_advance)
-                amounts_owed[email] -= drawdown_amount
+                amounts_payable[email] -= drawdown_amount
 
         # note that these are drawn down amounts and therefore have negative amounts
         # and that's why we take the absolute value here
         redistribution_pot += sum(abs(a.amount) for a in negative_advances)
 
-        # TODO: finally, redistribute the pot over all payable contributors
-        # go through attributions for payable people and redistribute pot accordingly
-        # create advances for each of those amounts, and add the amount to amounts_owed
-        # later, generate transactions from the final amounts in amounts_owed
-
-        # this will now need to be updated to reflect the above comment
-        # will be almost the same as before except that remaining_amount will be
-        # redistribution_pot instead.
-        equity_transactions = (distribute_remaining_amount(remaining_amount,
-                                                           truncated_payable_attributions,
-                                                           payment)
-                               if remaining_amount else [])
-
+        # redistribute the pot over all payable contributors - produce fresh advances and add to amounts owed
+        fresh_advances = redistribute_pot(redistribution_pot,
+                                          attributions,
+                                          unpayable_contributors,
+                                          payment.payment_file,
+                                          amounts_payable)
+        # TODO - generate transactions from the final amounts in amounts_payable
```

Comment:
This function has gotten pretty huge. Would be great to refactor it into helpers if possible. Just noting (not for this PR).

Reply:
Definitely